### PR TITLE
[FIXED JENKINS-28409] - Properly close parent streams in EnvInjectPasswordsOutputStream

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-28409

@reviewbybees

### CODE REVIEW

**Comment by jglick** [comment]
`EnvInjectBuildWrapperTest.testPropertiesContentCustomWorkspace` failure.

File: src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPasswordTest.java [hunk 1]
```diff
@@ -53,10 +62,103 @@ public void testEnvInjectJobParameterPassword() throws Exception {
...
         //The value must be encrypted in the envVars
         Assert.assertEquals(Secret.fromString(PWD_VALUE).getEncryptedValue(), envVars.get(PWD_KEY));
+    }    
+    
+    /**
+     * A wrapper for an {@link OutputStream}, which raises the flag when a
+     * stream gets closed externally.
+     * @since TODO
+     */
+    public static class FileLeakDetectorStream extends OutputStream {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPasswordTest.java]: Might be slightly easier to use a `FilterOutputStream`.

File: src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPasswordTest.java [hunk 2]
```diff
@@ -53,10 +62,103 @@ public void testEnvInjectJobParameterPassword() throws Exception {
...
+                new EnvInjectPasswordEntry(PWD_KEY, PWD_VALUE)
+        });
+        final FileLeakBuildWrapper fileLeakDetector = new FileLeakBuildWrapper();
+
+        project.getBuildWrappersList().add(fileLeakDetector);
+        project.getBuildWrappersList().add(passwordWrapper);
+        
+        @SuppressWarnings("deprecation")  
+        FreeStyleBuild build = project.scheduleBuild2(0, new Cause.UserCause()).get();
+        Assert.assertEquals(Result.SUCCESS, build.getResult());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPasswordTest.java]: Tip:

```java
buildAndAssertSuccess(project);
```

As intermediate steps, you can omit the `Cause`, use `assertBuildStatusSuccess`, etc.

**Comment by jglick** [comment]
:+1: assuming the test gets fixed.

**Comment by oleg-nenashev** [comment]
> EnvInjectBuildWrapperTest.testPropertiesContentCustomWorkspace failure.

I doubt it's related to the fix. EnvInject builds continuously fail due to such issue

**Comment by oleg-nenashev** [comment]
Fixed both improvement tips

**Comment by tfennelly** [comment]
@oleg-nenashev looks like the test failure is related to this change, right?

**Comment by oleg-nenashev** [comment]
@tfennelly 
No, I don't think so. This error appears in every pull request build against 1.91.x. Somebody should fix the test BTW

**Comment by tfennelly** [comment]
:+1: cool thanks for clarifying

**Comment by lanwen** [comment]
fixed test in #51

**Pull request merged**